pkg/flowaggregator/clickhouseclient/clickhouseclient.go

```diff
@@ -0,0 +1,511 @@
+	sourceNodeName                       string
+	destinationPodName                   string
+	destinationPodNamespace              string
+	destinationNodeName                  string
+	destinationClusterIP                 string
+	destinationServicePort               uint16
+	destinationServicePortName           string
+	ingressNetworkPolicyName             string
+	ingressNetworkPolicyNamespace        string
+	ingressNetworkPolicyRuleName         string
+	ingressNetworkPolicyRuleAction       uint8
+	ingressNetworkPolicyType             uint8
+	egressNetworkPolicyName              string
+	egressNetworkPolicyNamespace         string
+	egressNetworkPolicyRuleName          string
+	egressNetworkPolicyRuleAction        uint8
+	egressNetworkPolicyType              uint8
+	tcpState                             string
+	flowType                             uint8
+	sourcePodLabels                      string
+	destinationPodLabels                 string
+	throughput                           uint64
+	reverseThroughput                    uint64
+	throughputFromSourceNode             uint64
+	throughputFromDestinationNode        uint64
+	reverseThroughputFromSourceNode      uint64
+	reverseThroughputFromDestinationNode uint64
+}
+
+func NewClickHouseClient(input ClickHouseInput) (*ClickHouseExportProcess, error) {
+	dsn, err := input.getDataSourceName()
+	if err != nil {
+		klog.Errorf("error parsing ClickHouse DSN: %v", err)
+		return nil, err
+	}
+	connect, err := sql.Open("clickhouse", dsn)
+	if err != nil {
+		klog.Errorf("error when opening DB connection to %s: %v", input.DbURL, err)
+		return nil, err
+	}
+
+	// Test ping DB
+	if err = connect.Ping(); err != nil {
+		if exception, ok := err.(*clickhouse.Exception); ok {
+			klog.Errorf("error pinging DB: [%d] %s", exception.Code, exception.Message)
+		} else {
+			klog.Errorf("error pinging DB: %v", exception)
+		}
+		return nil, err
+	}
+
+	// Test open Transaction
+	tx, err := connect.Begin()
+	if err == nil {
+		_, err = tx.Prepare(insertQuery)
+	}
+	if err != nil {
+		klog.Errorf("error when preparing insert statement: %v", err)
+		return nil, err
+	}
+	_ = tx.Commit()
+
+	chClient := &ClickHouseExportProcess{
+		db:       connect,
+		dsn:      dsn,
+		deque:    deque.New(),
+		mutex:    sync.RWMutex{},
+		stopChan: make(chan bool),
+	}
+	chClient.queueSize = maxQueueSize
+
+	return chClient, nil
+}
+
+func (ch *ClickHouseExportProcess) CacheSet(set ipfixentities.Set) {
+	record := set.GetRecords()[0]
+	chRow := ch.getClickHouseFlowRow(record)
+
+	ch.mutex.Lock()
+	defer ch.mutex.Unlock()
+	for ch.deque.Len() >= ch.queueSize {
+		ch.deque.PopFront()
+	}
+	ch.deque.PushBack(chRow)
+}
+
+func (ch *ClickHouseExportProcess) Start() {
+	go ch.flowRecordPeriodicCommit()
+	<-ch.stopChan
+}
+
+func (ch *ClickHouseExportProcess) Stop() {
+	close(ch.stopChan)
+}
+
+func (ch *ClickHouseExportProcess) getClickHouseFlowRow(record ipfixentities.Record) ClickHouseFlowRow {
```

Comment:
Please use pointer for the returned type and insert the pointer to the queue (or the proposed channel) as `ClickHouseFlowRow` contains many fields, otherwise it would be copied everytime it's assigned to a variable, introducing unnecessary overhead.